tamr_toolbox/filesystem/cloud.py

```diff
@@ -0,0 +1,76 @@
+from google.cloud import storage
+from mypy_boto3_s3.client import S3Client
```

Comment:
Because these are optional dependencies, they should be imported in a slightly different way:
```suggestion
# Building our documentation requires access to all dependencies, including optional ones
# This environments variable is set automatically when `invoke docs` is used
BUILDING_DOCS = os.environ.get("TAMR_TOOLBOX_DOCS") == "1"
if BUILDING_DOCS:
    # Import relevant optional dependencies
    from google.cloud import storage
    from mypy_boto3_s3.client import S3Client
```

This also means references to these modules in type-hints need to be in double quotes. See my other comment.